temoa_model/temoa_rules.py

```diff
@@ -1412,6 +1412,66 @@ def ReserveMargin_Constraint(M, p, s, d):
 
     return cap_avail >= cap_target
 
+def OperReserveMargin_Constraint(M, p, s, d):
+    r"""
+
+During each period :math:`p`, the sum of the available operating capacity of all reserve
+technologies :math:`\sum_{t \in T^{e}} \textbf{OPCAPAVL}_{p,t}`, which are
+defined in the set :math:`\textbf{T}^{e}`, should exceed the load by
+:math:`OPRES`, the regional operational reserve margin. 
+
+.. math::
+   :label: op_reserve_margin
+
+       \sum_{t \in T^{e}} {
+       CC_t \cdot
+       \textbf{OPCAPAVL}_{p,t} \cdot
+       SEG_{s^*,d^*} \cdot C2A_t }
+       \geq
+       \sum_{ t \in T^{e},V,I,O } {
+           \textbf{FO}_{p, s, d, i, t, v, o}  \cdot (OPRES)
+       }
+
+       \\
+       \forall \{p, s, d\} \in \Theta_{\text{OpReserveMargin}}
+
+"""
+    if not M.tech_reserve:  # If reserve set empty, skip the constraint
+        return Constraint.Skip
+
+    opcap_total = sum(
+        value(M.OpResCapacityCredit[p, t])*
+        M.V_CapacityAvailableByPeriodAndTech[p, t]
+        * value(M.CapacityToActivity[t])
```

Comment:
This is a minor comment. I suggest removing value(M.CapacityToActivity[t]) and value(SegFrac[s,d]) from opcap_total, and dividing opcap_running and total_generation by them. This makes all the terms be in the form of GW.